### ISSUE TITLE
Unknown directive 'k8s_gateway' in coredns

### ISSUE THREAD

**Issue by macourteau** [ori-edge/k8s_gateway#76]
I've installed `k8s_gateway` using the helm chart per the instructions [here](https://github.com/ori-edge/k8s_gateway#install).

It seems to work as expected, resolving ingresses for the first zone configured (it doesn't resolve ingresses for other zones though, which is a separate issue, but any help on that topic too would be greatly appreciated), with a straightforward config like this:
```
apiVersion: v1
data:
  Corefile: |
    .:53 {
        errors
        health {
           lameduck 5s
        }
        ready
        kubernetes cluster.local in-addr.arpa ip6.arpa {
           pods insecure
           fallthrough in-addr.arpa ip6.arpa
           ttl 30
        }
        k8s_gateway internal.foo.com internal.bar.com
        prometheus :9153
        forward . /etc/resolv.conf {
           max_concurrent 1000
        }
        cache 30
        loop
        reload
        loadbalance
    }
kind: ConfigMap
```

However, when I restart coredns, it enters into a crashloop. Logs for the coredns pod say:
```
$ kubectl logs -n kube-system coredns-5d9bb78459-wxwjq
/etc/coredns/Corefile:12 - Error during parsing: Unknown directive 'k8s_gateway'
```

If I remove the `k8s_gateway` directive, and restart the crashlooped coredns, it is able to start successfully.

I must be doing something wrong, but I'm not quite sure what's going on. Any help appreciated!

**Comment by networkop** [comment]
I think you're confusing the two coredns's running in your cluster. If you install `k8s_gateway`  from helm chart, you won't get the `kubernetes` plugin included (see https://github.com/ori-edge/k8s_gateway#description)
How did you create this config file? did you edit it manually? It seems like you're restarting the in-cluster coredns plugin, instead of `k8s_gateway` and because `k8s_gateway` is an external plugin, it doesn't know how to interpret it (see https://coredns.io/manual/explugins/)

**Comment by macourteau** [comment]
Aha, interesting, you're absolutely right. Thanks for the explanation, everything makes so much more sense now. I didn't realize there was a separate config for the `k8s_gateway` installed via the helm chart. I've edited that config instead, and that fixed all the issues I was having - thanks!

I think I got in that situation by mixing information from various sources, some of which were setting up a cluster without coredns and then installing coredns manually (e.g. [here](https://www.reddit.com/r/homelab/comments/ipsc4r/howto_k8s_metallb_and_external_dns_access_for/)), without fully understanding the distinction between the coredns instance running in the cluster for external name resolution vs the in-cluster instance.
